src/extract/gatherInitialSources.js

```diff
@@ -41,14 +42,19 @@ function gatherScannableFilesFromDir (pDirName, pOptions) {
 module.exports = (pFileDirArray, pOptions) => {
     const lOptions = Object.assign({baseDir: process.cwd()}, pOptions);
 
-    return pFileDirArray.reduce(
-        (pAll, pThis) => {
-            if (fs.statSync(path.join(lOptions.baseDir, pThis)).isDirectory()) {
-                return pAll.concat(gatherScannableFilesFromDir(pThis, lOptions));
-            } else {
-                return pAll.concat(pThis);
-            }
-        },
-        []
-    );
+    return pFileDirArray
+        .reduce(
+            (pAll, pThis) => pAll.concat(glob.sync(pThis, {cwd:lOptions.baseDir})),
```

Comment:
performance tip: use the same options object for each call to glob and initialize the cache properties:

```js
const lGlobOptions = {
    cwd: lOptions.baseDir,
    cache: {},
    symlinks: {},
    statCache: {},
    realpathCache: {},
};
glob.sync(pFileDirArray[0], lGlobOptions);
glob.sync(pFileDirArray[1], lGlobOptions); // will use the cache of the previous call
```

Reply:
hmm, indeed. Didn't properly disable globbing in my manual tests :-/. Fix in 3804895 . I'm looking for a way to automate that test (at least locally).

Thanks for the performance optimisation tip; I'll look into it (measure the difference & if it's better put it in).